nlp_primitives/number_of_common_words.py

```diff
@@ -0,0 +1,53 @@
+from typing import Iterable
+
+import pandas as pd
+from featuretools.primitives.base import TransformPrimitive
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import IntegerNullable, NaturalLanguage
+
+from .constants import common_words_1000
+
+
+class NumberOfCommonWords(TransformPrimitive):
+    """Determines the number of common words in a string.
+    Description:
+        Given list of strings, determine the number of words that appear in a supplied wordbank.
```

Comment:
Probably would be good to clarify that the word bank is optional and mention what gets used by default if one is not supplied.